esp-hal/src/uart.rs

```diff
@@ -840,13 +843,21 @@ where
             | RxEvent::FrameError
             | RxEvent::ParityError;
         let events = self.uart.info().rx_events().intersection(errors);
-        let result = rx_event_check_for_error(events);
+        let result = self.handle_rx_events(events);
         if result.is_err() {
             self.uart.info().clear_rx_events(errors);
         }
```

Comment:
The sequence of events here is:
1. Reset FIFO.
2. Clear FIFO interrupt.

My concern here is this code getting interrupt in between these two operations.
How does the UART hardware know when it can start writing to the RX FIFO again?

If you reset the FIFO, then the code gets interrupted to do something else (WiFi or defmt), then the FIFO overflows again, the code resumes, then clears the fifo overflow interrupt.
You end up in a situation where you still need to reset the FIFO but haven't.

Reply:
good point